pyart/io/output_to_geotiff.py

```diff
@@ -0,0 +1,246 @@
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import os
+try:
+    from osgeo import gdal
+    IMPORT_FLAG = True
+except ImportError:
+    IMPORT_FLAG = False
+
+
+def write_grid_geotiff(filename, grid, field=None, rgb=False, level=None,
+                       cmap='viridis', vmin=0, vmax=75, color_levels=None,
+                       warp=False, sld=False):
+    """
+    Write a Py-ART Grid object to a GeoTIFF file. The GeoTIFF can be the
+    standard Azimuthal Equidistant projection used in Py-ART, or a lat/lon
+    projection on a WGS84 sphere. The latter is typically more usable in web
+    mapping applications. The GeoTIFF can contain a single float-point raster
+    band, or three RGB byte raster bands. The former will require an SLD file
+    for colorful display using standard GIS or web mapping software, while the
+    latter will show colors "out-of-the-box" but lack actual data values. The
+    function also can output an SLD file based on the user-specified inputs.
+    User can specify the 2D vertical level to be output. If this is not
+    specified, a 2D composite is created. User also can specify the field
+    to output.
+
+    This function requires GDAL Python libraries to be installed. These are
+    available via conda; e.g., 'conda install gdal'
+
+    Parameters
+    ----------
+    filename : str
+        Filename to save grid to
+    grid : Grid
+        Grid object to write
+    rbg : Boolean
+        True - Output 3-band RGB GeoTIFF
+        False - Output single-channel, float-valued GeoTIFF. For display,
+                likely will need an SLD file to provide a color table.
+    level: int
+        Index for z-axis plane to output. None gives composite values
+        (i.e., max in column)
+    cmap : str or matplotlib colormap object
+        Colormap to use for RGB output or SLD file
+    vmin : int or float
+        Minimum value to color for RGB output or SLD file
+    vmax : int or float
+        Maximum value to color for RGB output or SLD file
+    color_levels : int
+        Number of color levels in cmap. Useful for categorical colormaps
+        with steps << 255 (e.g., hydrometeor ID).
+    warp : Boolean
+        True - Use gdalwarp (called from command line using os.system)
+               to warp to a lat/lon WGS84 grid
+        False - No warping will be performed. Output will be Az. Equidistant.
+    sld : Boolean
+        True - Create a Style Layer Descriptor file (SLD) mapped to vmin/vmax
+               and cmap. File is named same as output TIFF, except for .sld
+               extension.
+        False - Don't do this
+    """
+    if not IMPORT_FLAG:
+        print('GDAL not detected, GeoTIFF output failure!')
+        return
+
+    # Allow the user to swap filename/grid position in call
+    if type(filename) is not str:
+        fname = grid
+        gridvol = filename
+    else:
+        fname = filename
+        gridvol = grid
```

Comment:
I understand the idea behind this but I do not like it.  Either a user gives the parameter in the correct order or the function should fail.